pandas/tests/indexes/ranges/test_range.py

```diff
@@ -272,6 +272,24 @@ def test_cache(self):
         expected = np.arange(0, 100, 10, dtype="int64")
         tm.assert_numpy_array_equal(idx._cache["_data"], expected)
 
+    def test_cache_after_calling_loc_with_array(self):
+        # GH 53387
+        # the cache will contain a _constructor key, so it should be tested separately
+        idx = RangeIndex(0, 100, 10)
+        df = pd.DataFrame({"a": range(10)}, index=idx)
```

Comment:
is the dataframe part of this necessary?  not a huge deal but in general we try to keep DataFrame/Series out of tests/indexes/

Reply:
This dataframe is defined in the same way as the one in `test_cache` at https://github.com/pandas-dev/pandas/blob/4d38764f25444c0e61401e583cb684d75ac858a3/pandas/tests/indexes/ranges/test_range.py#L249
To avoid instantiating a new dataframe, I can move these 3 checks into the same test, with a few minor changes.